### ISSUE TITLE
Exceptions cause a filterworker to die

### ISSUE THREAD

**Issue by jordansissel** [logstash-plugins/logstash-filter-ruby#2]
An exception (NameError, NoMethodError, etc) will cause the filterworker to die off leaving basically nothing reading from the input-to-filter queue.

Should be  easy to fix, and we should include a test case for this as well.

---

On error, the filter worker should continue like normal. Perhaps we should add a tag _rubyexception?

---

Example

```
% bin/logstash -e 'filter { ruby { code => "whatever" } }'
hello world
Using milestone 1 filter plugin 'ruby'. This plugin should work, but would benefit from use by folks like you. Please let us know if you find bugs or have suggestions on how to improve this plugin.  For more information on plugin milestones, see http://logstash.net/docs/2.0.0.dev/plugin-milestones {:level=>:warn}

Exception in filterworker {"exception"=>#<NameError: undefined local variable or method `whatever' for <LogStash::Filters::Ruby code=>"whatever">:LogStash::Filters::Ruby>, "backtrace"=>["(ruby filter code):1:in `register'", "org/jruby/RubyProc.java:271:in `call'", "/Users/jls/projects/logstash/vendor/bundle/jruby/1.9/gems/logstash-filter-ruby-0.1.2/lib/logstash/filters/ruby.rb:38:in `filter'", "(eval):50:in `initialize'", "org/jruby/RubyArray.java:1613:in `each'", "org/jruby/RubyEnumerable.java:805:in `flat_map'", "(eval):46:in `initialize'", "org/jruby/RubyProc.java:271:in `call'", "/Users/jls/projects/logstash/lib/logstash/pipeline.rb:272:in `filter'", "/Users/jls/projects/logstash/lib/logstash/pipeline.rb:206:in `filterworker'", "/Users/jls/projects/logstash/lib/logstash/pipeline.rb:150:in `start_filters'"], :level=>:error}
```

**Comment by bradvido** [comment]
I have this problem as well. I have a ruby block that called `to_f` on a filed that was array (but i expected to be a single string), which caused `NoMethodError`. The filter thread died and never restarted, so logstash completely stopped processing events. This should be documented at the very least, but it would be better if this filter caught unhandled exceptions and maybe added a tag, similar to the `_grokparsefailure` tag.

**Comment by jsvd** [comment]
this has been implemented in #5
